File: Solutions/Python/1979.py
class Solution:
    def findGCD(self, nums: List[int]) -> int:
        com = []
        maxi = max(nums)
        mini = min(nums)
        for i in range(1, maxi+1):
            if maxi%i == 0 and mini%i == 0:
                com.append(i)
        return max(com)

File: Solutions/Java/1979.java
class Solution {
    public int findGCD(int[] nums) {
        int min = Integer.MAX_VALUE, max = Integer.MIN_VALUE;

        for(int e : nums) {
            min = Math.min(e, min);
            max = Math.max(e, max);
        }

        while(min > 0) {
            int temp = min;
            min = max % min;
            max = temp;
        }
        return max;
    }
}
